### DISCUSSION TITLE
An optional landing page

### DISCUSSION THREAD

**Discussion by christianavi** [retypeapp/retype#548] [Ideas]
I think it would be great to have an optional landing page as well. Could be just a Markdown. I am not sure though how it's gonna be implemented and what changes are required if this was to happen.

**Comment by geoffreymcgill** [comment]
Thanks for the feature request. This is functionality we have wanted for a long time and at some point we will be implementing. At the moment, you can create a custom `/index.html` page for your project home page, although you would 100% control the HTML. 

You would need to remove your projects `readme.md` or `index.md` page in that scenario.

**Comment by finkarder** [comment]
This is something I would really welcome as well. 

Having `landing page` + `docs` + `blog` in one deploy would be great. But given experience with Docusaurus I'm concerned about the possible added complexity. How to support compelling landing pages using Markdown for tech-savvy non-devs ?

As a new user I was unaware of the `index.html` option (could be me). Perhaps until you resolve how to support landing pages this option could be made more obvious in the docs and/or use it on the retype site itself.

**Comment by geoffreymcgill** [comment]
> But given experience with Docusaurus I'm concerned about the possible added complexity. How to support compelling landing pages using Markdown for tech-savvy non-devs ?

We would likely handle using the Page [`layout`](https://retype.com/configuration/page/#layout) config. Maybe something as simple as `layout: home`. That could even be automated to some degree, where Retype would just treat the root `readme.md` or `index.md` file as the _home page_ and automatically set the `layout`. Similar to how any pages within the `/blog` folder automatically get assigned the `layout: blog` layout.

Ideally we would develop several landing page options that you could switch between by just changing the `layout` value. 

A few new components would be ideal too. 

This is all on my list of features to implement. Just need more time, or funds to hire another engineer & designer.

> But given experience with Docusaurus I'm concerned about the possible added complexity.

@rab-dev Can you explain this comment further? What has been your experience with Docusaurus? In general, from my perspective, they appear to be making a good product. Although, I have to admit, I've never actually implemented a full project using Docusaurus, so I've never been too deep.

Hope this helps.

**Comment by finkarder** [comment]
It's not really about going deep with Docusaurus and more about wanting to stay shallow. Docusaurus supports MDX / JSX / React so you can do anything & get as fancy as you like. The results look great but it's over-complicated imo. I don't want to write code to write docs, I  just want to write docs. When a product becomes bloated it becomes over-complicated for non-experts purely by them not being able to see the wood for the trees. You could argue just use a subset of Docusaurus but I say keep it simple in the first place if you can. 

For landing pages you may be able to do something useful around markup for "sections". I've had simple 2-column sections before where the background color & column positions automatically alternated. Could be useful for landing pages.

Thanks for your help.

**Comment by geoffreymcgill** [comment]
@rab-dev Excellent feedback. Thanks for taking the time to respond.

> It's not really about going deep with Docusaurus and more about wanting to stay shallow. 
> I don't want to write code to write docs, I just want to write docs.

Very well said, and I agree 100%.

**Comment by markbattistella** [comment]
@geoffreymcgill this would be great too with the new `visibility` levels too. Being able to set a landing page with login required, then access to the whole site would be nice for public sites but private information

**Comment by finkarder** [comment]
I played with this and created a custom `/index.html` in the root src folder.

It raised a question about pathnames.

I would like option A, but I can only seem to get B or C with just retype.
I assume I could get D if I introduce an extra build tool / step.

Could be my limited understanding of retype but if anyone knows how to get Option A then we would have `landing page` + `docs` + `blog` from just retype (albeit with an externally created landing page).

||A|B|C|D|
|--|--|--|--|--|
| landing page |`example.com`|`example.com`|`example.com/docs`|`example.com`|
| docs               |`example.com/docs`|`example.com`|`example.com/docs`|`example.com/docs`|
| blog               |`example.com/blog`|`example.com/blog`|`example.com/docs/blog`|`example.com/docs/blog`|

**Comment by geoffreymcgill** [comment]
Hi @rab-dev,

I was able to get option A working a couple different ways, although I'm not 100% sure if this covers your requirements.

https://github.com/retypeapp/retype/assets/62210/85d2b900-41ce-4a0d-8194-6a02a98e5612

The above uses the [`route`](https://retype.com/configuration/page/#route) config to adjust the URLs. For example, using the following file structure:

<img width="350" alt="Screenshot 2023-08-26 at 3 10 40 AM" src="https://github.com/retypeapp/retype/assets/62210/e75a09d5-e8cd-4e1a-804e-781a19866965">

Download .zip: [home-sample.zip](https://github.com/retypeapp/retype/files/12444976/home-sample.zip)

Here are a few of the files from that sample:

#### /guides/index.yml

```yml
route: /docs/
```

#### /docs.md

```md
---
order: 100
---
# Docs Home

Welcome to the documentation!
```

You could add a `faq.md` page into the root of the project and still maintain the `/docs` routing by using the following `route` config:

<img width="350" alt="Screenshot 2023-08-26 at 3 13 01 AM" src="https://github.com/retypeapp/retype/assets/62210/b723de10-a27a-40a9-b42b-b2a9ccba0e02">

#### /faq.md

```
---
route: /docs/faq/
---
# FAQ
```

---

Another basic option would be to just add all the documentation pages into a `/docs/` folder, then no `route` configs would be required. Although, the left tree-nav would show a root **Docs** folder. 

Download .zip: [home-sample-no-route.zip](https://github.com/retypeapp/retype/files/12444971/home-sample-no-route.zip)

<img width="352" alt="Screenshot 2023-08-26 at 3 17 14 AM" src="https://github.com/retypeapp/retype/assets/62210/ec0fa0ad-e75d-46cc-bdef-bb1fb2a67000">

<img width="769" alt="Screenshot 2023-08-26 at 3 22 36 AM" src="https://github.com/retypeapp/retype/assets/62210/a83f8020-1e34-4944-aeac-fa1e98db0eb5">

---

Hope this helps.

**Comment by finkarder** [comment]
Excellent!

Thank you for this. It is exactly the result I was looking for.

Attached is a simple extension to your solution showing how I have integrated my `index.html` into the mix. 

I've actually used an `index.md` which then includes `home.html` so the regular retype navbar still appears on the homepage.  And I modified some retype css settings to get a full-width homepage.

I admit it's a hack but it works for me.

**Download** [home-docs-blog.zip](https://github.com/retypeapp/retype/files/12446754/home-docs-blog.zip)

![home-docs-blog](https://github.com/retypeapp/retype/assets/61521617/a095324a-1285-429b-b802-57dba1647da1)
